Feature/22 add x and y labels

GegznaV opened the pull request:
Closes #22 

I think, some unit tests should be added for this functionality, but I do not have time for this now. @bryanhanson, please review and give feedback.

GegznaV committed:
Create helper jdx_extract_value()

GegznaV committed:
Add labels for spc and wl

bryanhanson reviewed (changes requested):
Looks good overall. It looks like `jdx_extract_value` can be used not only as you have it, but also for CB's desire to extract arbitrary keys (in the long run).  However, I think the following line:
`key_pattern <- paste0("##\\$?", key, "=")`
should be changed to:
`key_pattern <- paste0("##\\$?", key, "=\\s*")`
to account for the fact that some vendors put variable amounts of space after the =.

In general this grep pattern will likely catch any of the standard fields along with some of the vendor fields, but I took a quick look at some of the 100+ JCAMP files I have in my test suite and there are other patterns in use -- many -- I would have to set aside a good block of time to figure out a more general approach, and this can wait.

GegznaV commented:
> to account for the fact that some vendors put variable amounts of space after the =.

For that, I used `trimws()`

GegznaV commented:
>  but I took a quick look at some of the 100+ JCAMP files I have in my test suite and there are other patterns in use 

Yes, I saw only 2-3 patterns. And this function is not general enough to account for all possible cases. I wrote the function by keeping `XUNITS` and `YUNITS` in mind only without looking to other possible usecases.

GegznaV committed:
Extend pattern to take whitespace into account

GegznaV commented:
> should be changed to:
> `key_pattern <- paste0("##\\$?", key, "=\\s*")`

Done.

GegznaV committed:
Return named vector

GegznaV committed:
Allow extraction of several keys

GegznaV committed:
Make optionally named output

GegznaV committed:
Fix typo

GegznaV committed:
Do not use extraction of several kays and naming

... as in certain situations they are unreliable and give misleading results or result in unwanted errors.

GegznaV committed:
Update from `devel`

GegznaV committed:
Remove duplacated code

Related to #27

GegznaV committed:
Add labels to `hyperSpec` object

GegznaV committed:
Add comment

GegznaV committed:
Fix typos

GegznaV committed:
Update style

GegznaV committed:
Add unit tests

GegznaV committed:
Updae from

GegznaV commented:
@bryanhanson, ready for review again.

GegznaV commented:
The message of 6ef0ed2 fad to be `Update from devel`

cbeleites commented on R/read_jdx.R:
```diff
@@ -79,6 +74,10 @@ read_jdx <- function(file = stop("filename is needed"), SOFC = TRUE, debug = 0)
+
+  # Output
+  .spc_io_postprocess_optional(spc, filename = file)
+  # return(list(metadata = list_jdx[[2]], hyperSpec = spc))
```
please remove this commented-out line of code

GegznaV replied:
I just moved the two rows outside the if-else statement. I'm not sure why the commented code was left. But, in general, I agree that there should not be any commented code left in the packages.

GegznaV replied:
Done.

bryanhanson reviewed (approved):
With CB's note to remove the commented line, this looks good to me.

GegznaV committed:
Remove the commented code

GegznaV committed:
Merge branch 'develop' into feature/22-add-labels--x-y
